pos_pricelist/static/src/js/models.js

```diff
@@ -418,12 +465,8 @@ function pos_pricelist_models(instance, module) {
             var items = [], i, len;
             for (i = 0, len = db.pricelist_item_sorted.length; i < len; i++) {
                 var item = db.pricelist_item_sorted[i];
-                if ((item.product_id === false
-                    || item.product_id[0] === product.id) &&
-                    (item.categ_id === false
-                    || categ_ids.indexOf(item.categ_id[0]) !== -1) &&
-                    (item.price_version_id[0] === version.id)) {
-                    items.push(item);
+                if (this.rule_match(item, product, qty, version)) {
+                  items.push(item)
```

Comment:
If we only work with valid rules from here, can we break when we found the first valid rule here?